### ISSUE TITLE
Cause Zotero RSS Feeds stop work

### ISSUE THREAD

**Issue by CaoZihang** [PubPeerFoundation/pubpeer_zotero_plugin#38]
When enabling this plugin, Zotero(v.6.0.12)  RSS Feeds will stop work.

**Comment by retorquere** [comment]
Please select `Help` -> `Debug Output Logging` => `Restart with logging enabled`, and after Zotero has restarted, `Help` => `Send debug log to file.io`, then paste the ID you see on the screen here.

**Comment by CaoZihang** [comment]
D443216290

**Comment by retorquere** [comment]
That was not generated by "send to file.io". The log generated by "send to file.io" has a `+` in the ID.

**Comment by CaoZihang** [comment]
Sorry, there was no "send to file.io" choice in my Zotero. So I just signed up a file.io account and pasted the log to a txt file. There is the link https://file.io/ojWinim601SI, I didn't find the ID.

**Comment by retorquere** [comment]
You must have viewed the link -- files on file.io can only be viewed/downloaded once. Can you download and install https://github.com/retorquere/zotero-debug-log, reproduce the problem (again after restart with "Restart with logging enabled", and the "send to file.io"? This plugin provides file.io log submission separately.

**Comment by CaoZihang** [comment]
Someone viewed the link. The ID is 5y4uFVxgRsiE-RJHK7WJ6, but there is no +, so I also uploaded a link https://file.io/jDcTm7MUKN0Q.

**Comment by retorquere** [comment]
My bad, the dash is correct, not a plus.

**Comment by retorquere** [comment]
What exactly doesn't work with the RSS feeds? I see an error (among a few) for https://academic.oup.com/rss/site_5397/3258.xml, but that feed works for me with pubpeer (and BBT) installed.

**Comment by CaoZihang** [comment]
RSS Feeds can not update or add. There is a "!" signal on the Feeds icon (Pubpeer can be used at the same time). The error report mentioned "Error: Processing failed". I don't know whether or not it is an individual case, but I can repeat this problem reliably and use RSS Feeds successfully by disabling Pubpeer(Pubpeer stop work, of course).
![image](https://user-images.githubusercontent.com/64113110/185869162-0e69d075-7484-415b-bf99-9f38f0eeb401.png)

**Comment by dstillman** [comment]
Can you reproduce this with no other extensions except PubPeer installed?

**Comment by CaoZihang** [comment]
Yes, I can reproduce this with no other plugins.

**Comment by retorquere** [comment]
This is reproducible for me. The first suspect would be the monkey patches, but I don't see any monkey patches that BBT doesn't also make, and BBT doesn't trigger the issue. The error I see is:

```
(3)(+0000415): NS_ERROR_UNEXPECTED Exception: 

    _parseAndNotify@chrome://zotero/content/xpcom/feedReader.js -> resource://zotero/feeds/SAXXMLReader.js:76:15
    onResponseAvailable@chrome://zotero/content/xpcom/feedReader.js -> resource://zotero/feeds/SAXXMLReader.js:66:3
(3)(+0000000): FeedReader: Terminating feed reader (Processing failed)
(1)(+0000002): Error processing feed from http://feeds.hbr.org/harvardbusiness/:

Error: Processing failed
(2)(+0000000): Warning: Attempting to set unsaved feed property "_feedLastCheckError
```

but pubpeer doesn't touch those.

**Comment by retorquere** [comment]
It's not the monkey patches. I've disabled all monkey patches and the problem still occurs.

**Comment by retorquere** [comment]
I've narrowed it down to this:

```
  private dom = {
    parser: Components.classes['@mozilla.org/xmlextras/domparser;1'].createInstance(Components.interfaces.nsIDOMParser),
    serializer: Components.classes['@mozilla.org/xmlextras/xmlserializer;1'].createInstance(Components.interfaces.nsIDOMSerializer),
  }
```

does this ring a bell @dstillman? Simply the fact that these are created at any point (at all points I tested at least) triggers the problem. BBT uses `@xmldom/xmldom` rather than these native parser/serializers.

**Comment by retorquere** [comment]
Dropping in `@xmldom/xmldom` for the parser and serializer fixes the issue. Less efficient, but it isn't called a massive amount, so maybe acceptable.

**Comment by CaoZihang** [comment]
To my mind, it's acceptable from what you described. There are only a few papers that have pubpeer comments in my major, so I do not care about efficiency too much.

**Comment by dstillman** [comment]
It looks like it's the DOMParser line. This would be a bug, but I can't reproduce this in Zotero itself. E.g., if I run:

```
var parser = Components.classes['@mozilla.org/xmlextras/domparser;1'].createInstance(Components.interfaces.nsIDOMParser)
var serializer = Components.classes['@mozilla.org/xmlextras/xmlserializer;1'].createInstance(Components.interfaces.nsIDOMSerializer)
```
in the Run JavaScript window at Zotero startup, before any feeds are loaded, I can still refresh feeds.

There could be a difference here on macOS (which I'm testing on) vs. Windows/Linux — the feed reader does some [slightly weird things](https://github.com/zotero/zotero/blob/f4dfea6849455dd9c4eeb6fdb64f6a9ff2ad0ee0/chrome/content/zotero/xpcom/feedReader.js#L133-L165) to create a sandbox for the feed parser.

If you can reproduce this on stock Zotero, I can probably fix it.

**Comment by retorquere** [comment]
On Linux, if I enter https://academic.oup.com/rss/site_5397/3258.xml in the URL field for a new feed, the Title field does not get filled, and the Save button remains disabled, so I cannot currently test this scenario. I don't have a windows machine.

**Comment by dstillman** [comment]
Not sure what you mean there. Are you saying feed reading is broken for you in general? With all extensions disabled? We'd want a Debug ID for that — feed reading certainly isn't broken in general on Linux.

**Comment by retorquere** [comment]
No, I get this behavior with (only) the pubpeer plugin installed. On MacOS, I can still enter the new feed. If I disable pubpeer, add the feed, enable pubpeer, I get the same refresh problem as I get on MacOS.

**Comment by dstillman** [comment]
But if you can't create the feed, that's a feed-parsing problem — presumably the same one — and should be reported.

**Comment by dstillman** [comment]
And if you can only reproduce it with PubPeer installed, then I'm asking whether you can reproduce it with the manual commands in Run JavaScript, without PP installed.

**Comment by retorquere** [comment]
D566667533, but only with pubpeer installed. When I run via "Run Javascript", both entering a new feed and refreshing it works.

**Comment by CaoZihang** [comment]
D443216290



At 2022-08-21 22:21:35, "Emiliano Heyns" ***@***.***> wrote:

Please select Help -> Debug Output Logging => Restart with logging enabled, and after Zotero has restarted, Help => Send debug log to file.io, then paste the ID you see on the screen here.

**Comment by CaoZihang** [comment]
It works, thanks for your help!
